packages/opencensus-web-instrumentation-zone/src/interaction-tracker.ts

```diff
@@ -237,6 +242,47 @@ export class InteractionTracker {
     stopWatch.stopAndRecord();
     delete this.interactions[interactionId];
   }
+
+  // Monkey-patch `History API` to detect route transitions.
+  // This is necessary because there might be some cases when
+  // there are several interactions being tracked at the same time
+  // but if there is an user interaction that triggers a route transition
+  // while those interactions are still in tracking, only that interaction
+  // will have a `Navigation` name. Otherwise, if this is not patched, the
+  // other interactions will change the name to `Navigation` even if they
+  // did not cause the route transition.
+  private patchHistoryApi() {
+    const pushState = history.pushState;
+    history.pushState = (
+      //tslint:disable:no-any
+      data: any,
+      title: string,
+      url?: string | null | undefined
+    ) => {
+      pushState.call(history, data, title, url);
+      this.maybeUpdateInteractionName();
+    };
+
+    const replaceState = history.replaceState;
+    history.replaceState = (
+      //tslint:disable:no-any
+      data: any,
+      title: string,
+      url?: string | null | undefined
+    ) => {
+      replaceState.call(history, data, title, url);
+      this.maybeUpdateInteractionName();
+    };
+  }
+
+  private maybeUpdateInteractionName() {
+    console.log('location changed!');
+    // Route transition is detected, thus the current interaction name might change.
+    const rootSpan = getRootSpan(Zone.current);
+    if (rootSpan && rootSpan.name.startsWith('<')) {
```

Comment:
What does the `rootSpan.name.startsWith('<')` check for? Is that in case the user gave an explicit tag for the element?

That's an interesting question as to whether the navigation name or the element tag name "wins". At any rate, I think commenting to explain it would be good.